blocks-common/i-bem/__dom/i-bem__dom.js

```diff
@@ -806,12 +806,24 @@ var DOM = BEM.DOM = BEM.decl('i-bem__dom', {
             ctx = this.findElem(ctx);
         }
 
+        names = names.split(' ');
+
         var _self = this.__self,
-            selector = '.' +
-                names.split(' ').map(function(name) {
-                    return buildClass(_self._name, name, modName, modVal);
-                }).join(',.');
-        return findDomElem(ctx, selector);
+            keys = names.map(function(name) {
+                return _self.buildClass(name, modName, modVal);
+            }),
+            isSingleName = keys.length === 1,
+            selector = '.' + (isSingleName? keys[0] : keys.join(',.')),
+            res = findDomElem(ctx, selector);
+
+        // caching results if possible
+        (ctx.length === 1 && this.domElem.length === 1 && ctx[0] === this.domElem[0]) &&
```

Comment:
case: 
element is at other dom node and block has been destructed.
so `ctx.length === 1` - true
`this.domElem.length` - Cannot read property 'length' of undefined
change to `ctx.length === 1 && this.domElem && this.domElem.length === 1 `